torch/_torch_docs.py

```diff
@@ -3178,6 +3178,54 @@ def merge_dicts(*dicts):
     tensor([ 0.,  2.,  1.,  0.])
 """.format(**common_args))
 
+add_docstr(torch.histogram,
+           r"""
+histogram(input, bins=10, weights=None, min=0, max=0, density=False) -> Tensor
+
+Computes the histogram of a tensor.
+
+Args:
+    {input}
+    bins (int or 1D Tensor): If int, defines the number of bins. If Tensor,
+        defines a monotonically increasing sequence of bin edges, allowing
+        for non-uniform bins. Default: 10
+    min (float): lower end of the range (inclusive), ignored if bins is a Tensor
+    max (float): upper end of the range (inclusive), ignored if bins is a Tensor
+    weights (Tensor): optional, weight for each value in the input tensor.
```

Comment:
"(Tensor, optional)"

Reply:
Question about weights and type promotion: if weights is a float tensor will the resulting histogram be float-valued, too? Or must weights be an integer tensor?

Reply:
If weights is defined, the resulting histogram will have the same dtype as weights, otherwise it will be an integer tensor, just like with bincount.

Reply:
Done.